theano/tensor/nnet/tests/test_nnet.py

```diff
@@ -755,7 +759,8 @@ def test_get_rid_of_advanced_indexing_version_of_xent(self):
         for expr in expressions:
             # Verify the optimizer worked on the expressions
             f = theano.function([x, y], expr, mode=mode)
-            self.assertTrue(hasattr(f.maker.fgraph.outputs[0].tag, 'trace'))
+            assert check_stack_trace(
+                f, ops_to_check=crossentropy_softmax_argmax_1hot_with_bias)
```

Comment:
@nouiz The other nodes did not exist in the original graph. In that test case, they are not used (I think), in some real-life cases, they will replace other nodes created elsewhere, or be merged with them.